streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -920,6 +921,7 @@ private void closeNonAssignedSuspendedStandbyTasks() {
                 } catch (Exception e) {
                     log.error("{} Failed to remove suspended task standby {}", logPrefix, suspendedTask.getKey(), e);
                 } finally {
+                    task.closeStateManager(false);
```

Comment:
It does in trunk -- but not in `0.10.2`. If `task.close()` throws, `task.closeStateManager(true)` is never called and the lock is not release.